### PR TITLE
Classify maintained images and custom images

### PR DESCRIPTION
**Pull request by DaoDaoNoCode**
Design: https://marvelapp.com/prototype/6dcfc3h/screen/84876569
Update both frontend and backend to check the BYON images and show them in different groups.

### CODE REVIEW

**Review by dlabaj — APPROVED** [review]
/LGTM

File: jupyterhub_singleuser_profiles/ui/src/ImageForm/ImageForm.tsx [hunk 1]
```diff
@@ -129,10 +129,62 @@ const ImageForm: React.FC<ImageFormProps> = ({ userConfig, onValidImage }) => {
...
   };
 
+  const imageGroup = (imageList: ImageType[], isDefault: boolean) => (
+    <>
+      <div className="jsp-spawner__image-options__subtitle">
+        {isDefault ? 'Red Hat Maintained' : 'Custom Images'}
+      </div>
+      {isDefault ? null : (
+        <div className="jsp-spawner__image-options__subtitle-hint">
+          Manage custom images in Settings.
```
Comment (dlabaj) [inline jupyterhub_singleuser_profiles/ui/src/ImageForm/ImageForm.tsx:139]: you can remove this Manage custom images in Setttings.  We are not going to support it.

**Review by dlabaj — CHANGES_REQUESTED** [review]
LGTM other the removing "Manage custom images in Settings." you can remove it.

File: jupyterhub_singleuser_profiles/ui/src/ImageForm/ImageForm.tsx (hunk 1 above)
Comment (DaoDaoNoCode) [reply jupyterhub_singleuser_profiles/ui/src/ImageForm/ImageForm.tsx]: Updated.

**Review by dlabaj — APPROVED** [review]

**Review by dlabaj — APPROVED** [review]

**Review by dlabaj — APPROVED** [review]
lgtm

**Pull request merged**